### PR TITLE
config: avoid to use default dir if root is valid

### PR DESCRIPTION
**Pull request by sctb512**
In the default configuration for snapshotter we fill the cacheDir and LogDir:
```go
    logConfig := &c.LoggingConfig
    if len(logConfig.LogDir) == 0 {
		logConfig.LogDir = filepath.Join(c.Root, logging.DefaultLogDirName)
	}
    cacheConfig := &c.CacheManagerConfig
	if len(cacheConfig.CacheDir) == 0 {
		cacheConfig.CacheDir = filepath.Join(c.Root, "cache")
	}
```
But we do not fill these elements after loading the configuration from the `toml' file and parsing the arguments. This results in cacheDir and LogDir not being in snapshotter's RootDir.

### CODE REVIEW

File: cmd/containerd-nydus-grpc/main.go [hunk 1]
```diff
@@ -65,6 +66,17 @@ func main() {
 				}
 			}
 
+			if snapshotterConfig.Root != "" {
+				logConfig := &snapshotterConfig.LoggingConfig
```
[the remark this answers is not in the document]
Comment (sctb512) [reply cmd/containerd-nydus-grpc/main.go:70]: I think `FillUpWithDefaults` might not work. But putting the implementation in `config.MergeConfig` makes more sense.

**Review by changweige — COMMENTED** [review]
In fact, cache dir and log dir do not have to locate under snapshotter's root. It can be anywhere as per the user's configurations.

**Comment by sctb512** [comment]
> In fact, cache dir and log dir do not have to locate under snapshotter's root. It can be anywhere as per the user's configurations.

Got it. Should we keep the cache dir and log dir under the root directory if the user does not specify these values.
The problems I encountered was that the cache dir and log dir are always in `/var/lib/containerd-nydus` even I specify a root directory.

**Comment by changweige** [comment]
> > In fact, cache dir and log dir do not have to locate under snapshotter's root. It can be anywhere as per the user's configurations.
> 
> Got it. Should we keep the cache dir and log dir under the root directory if the user does not specify these values. 

Yes. We should always put the log dir and cache dir under custom root dir

File: config/config.go [hunk 2]
```diff
@@ -220,6 +222,19 @@ func MergeConfig(to, from *SnapshotterConfig) error {
...
+		logConfig := &to.LoggingConfig
+		if logConfig.LogDir == "" {
+			logConfig.LogDir = filepath.Join(to.Root, logging.DefaultLogDirName)
+		}
+		cacheConfig := &to.CacheManagerConfig
+		if cacheConfig.CacheDir == "" {
+			cacheConfig.CacheDir = filepath.Join(to.Root, "cache")
+		}
+	}
+
```
Comment (changweige) [inline config/config.go:237]: Could we move the above code into `ProcessConfigurations`? I think `MergeConfig` is only a thin layer to merge default configurations and custom configurations. But we can tweak the merged configurations in `ProcessConfigurations`

**Comment by changweige** [comment]
Need rebase

File: config/config.go (hunk 2 above)
Comment (sctb512) [reply config/config.go]: Done. PTAL

**Review by changweige — APPROVED** [review]
LGMT, thanks

**Pull request merged**